### ISSUE TITLE
targetSdkVersion 23  lead crash

### ISSUE THREAD

**Issue by ghalong** [facebook/react-native#10454]
### Issue Description

 I introduce RN to my project,when I start up my App,it's crashed and I got the bellow info(Code Snippets) ,even I start up the sample of RN,if I change the targetSdkVersion to 23,the app also crashed.

I have cost a day and finally found it is targetSdkVersion 23 lead this crash,then I change the targetSdkVersion to 22,it's been OK,I don't know why.
### Steps to Reproduce / Code Snippets

[`java.lang.RuntimeException: Could not get BatchedBridge, make sure your bundle is packaged correctly
                                                            at com.facebook.react.cxxbridge.CatalystInstanceImpl.loadScriptFromAssets(Native Method)
                                                            at com.facebook.react.cxxbridge.JSBundleLoader$1.loadScript(JSBundleLoader.java:33)
                                                            at com.facebook.react.cxxbridge.CatalystInstanceImpl.runJSBundle(CatalystInstanceImpl.java:177)
                                                            at com.facebook.react.XReactInstanceManagerImpl$4.call(XReactInstanceManagerImpl.java:918)
                                                            at com.facebook.react.XReactInstanceManagerImpl$4.call(XReactInstanceManagerImpl.java:911)
                                                            at com.facebook.react.bridge.queue.MessageQueueThreadImpl$1.run(MessageQueueThreadImpl.java:74)
                                                            at android.os.Handler.handleCallback(Handler.java:739)
                                                            at android.os.Handler.dispatchMessage(Handler.java:95)
                                                            at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
                                                            at android.os.Looper.loop(Looper.java:158)
                                                            at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:196)
                                                            at java.lang.Thread.run(Thread.java:818)

`
and
`                                                         android.view.WindowManager$BadTokenException: Unable to add window android.view.ViewRootImpl$W@2e3a19b -- permission denied for this window type
                                                             at android.view.ViewRootImpl.setView(ViewRootImpl.java:875)
                                                             at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:337)
                                                             at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:91)
                                                             at android.app.Dialog.show(Dialog.java:350)
                                                             at com.facebook.react.devsupport.DevSupportManagerImpl.handleReloadJS(DevSupportManagerImpl.java:644)
                                                             at com.facebook.react.XReactInstanceManagerImpl$3$1.run(XReactInstanceManagerImpl.java:405)
                                                             at android.os.Handler.handleCallback(Handler.java:739)
                                                             at android.os.Handler.dispatchMessage(Handler.java:95)
                                                             at android.os.Looper.loop(Looper.java:158)
                                                             at android.app.ActivityThread.main(ActivityThread.java:7237)
                                                             at java.lang.reflect.Method.invoke(Native Method)
                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)`]
#### Expected Results

[targetSdkVersion 23  should not lead crash]
### Additional Information
- React Native version: [0.35.0]
- Platform(s) (iOS, Android, or both?): [Android 6.0.1  Samsung SM-G9200]
- Operating System (macOS, Linux, or Windows?): [macOS]

**Comment by leeight** [comment]
```
android.view.WindowManager$BadTokenException: Unable to add window android.view.ViewRootImpl$W@2e3a19b -- permission denied for this window type
```

**Comment by markspolakovs** [comment]
@ghalong "Could not get BatchedBridge, make sure your bundle is packaged correctly" seems like your Packager isn't running properly. If you're running a development build with the packager open, try running `adb reverse tcp:8081 tcp:8081` on your computer. If you're using it on your phone without the Packager, make sure you're running a production APK (`cd android && gradlew assembleRelease`).

**Comment by ghalong** [comment]
@markspolakovs Hi thank you,I am running the development build with the packager open and I also have run `adb reverse tcp:8081 tcp:8081` in my computer,but it also crashed as I describe above.

**Comment by ghalong** [comment]
@leeight maybe the RN team could fix the issues : ).

**Comment by SandroMachado** [comment]
@ghalong add the overlay permission and request it (or manually enable it).

**Comment by brentvatne** [comment]
@ghalong - a PR is welcome! as you can imagine, there are plenty of tasks to work on, so your help would be appreciated

**Comment by makenova** [comment]
@SandroMachado do you have an example of how to request the overlay permission?
I tried to use the `PermissionsAndroid` module and my app still crashes.

**Comment by SandroMachado** [comment]
@makenova I did it manually.

![baconmhc19qsandromachado11082016181656](https://cloud.githubusercontent.com/assets/631739/20111297/b766d832-a5df-11e6-8e18-e3649b83dd3a.gif)

**Comment by jpshelley** [comment]
PR Open to fix this ^^ #11316

**Comment by hramos** [comment]
Closing this issue because it has been inactive for a while. If you think it should still be opened let us know why.
